### PR TITLE
Publish version `0.1.0a0`

### PR DESCRIPTION
**Pull request by davidorme**
# Description

This PR is to merge the current `develop` into the release branch `main` so that we can then publish the current code onto `PyPI` and get the project name fixed in `PyPI`. As part of this process, I'll also run a test publish to `TestPyPI`.

This isn't a particular target hit or goal achieved, but it is worth making sure we have everything set up and have an official PyPI page for the upcoming paper submission.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [ ] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [ ] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md
Comment (dalonsoa) [inline PULL_REQUEST_TEMPLATE.md:1]: I believe this should be - or typically is - within the `.github` folder.

File: dependabot.yml
Comment (dalonsoa) [inline dependabot.yml:1]: This as well, within the `.github` folder.

**Review by dalonsoa — APPROVED** [review]
Looks good to me. I've just made a couple of suggestions to move files from one place to another. 

To automate the pubisling to PyPI when you create a new release, I'd suggest you follow something similar to what we implemented in this other project: https://github.com/EnergySystemsModellingLab/MUSE_OS/blob/develop/.github/workflows/publish.yml
You will need to setup accounts in PyPI and TestPyPI firs, obviously.

**Comment by davidorme** [comment]
@dalonsoa The PR template is in a recognised location, but I agree it's neater in `.github`. It seems `dependabot.yml` _must_ be in `.github` though, so I botched a PR two years ago 😄.

Having an automatic PyPI setup would be good - I might just do this one manually to get it released and then we can revisit that for the next release?

**Comment by dalonsoa** [comment]
Sounds good!

**Review by vgro — APPROVED** [review]
I have nothing to add :-)

**Comment by davidorme** [comment]
Just pulling in the `progress` update from #403.

**Pull request merged**